klog_test.go

```diff
@@ -1447,6 +1447,100 @@ func TestErrorSWithLogr(t *testing.T) {
 	}
 }
 
+func TestCallDepthLogr(t *testing.T) {
+	logger := &callDepthTestLogr{}
+	logger.resetCallDepth()
+
+	testCases := []struct {
+		name  string
+		logFn func()
+	}{
+		{
+			name:  "Info log",
+			logFn: func() { Info("info log") },
+		},
+		{
+			name:  "InfoDepth log",
+			logFn: func() { InfoDepth(0, "infodepth log") },
+		},
+		{
+			name:  "InfoSDepth log",
+			logFn: func() { InfoSDepth(1, "infoSDepth log") },
```

Comment:
Why we pass different depth into `InfoDepth` and `InfoSDepth`?

Reply:
Yeah, I think this is showing that the current implementation is slightly incorrect

We need to also apply this patch:

```
diff --git a/klog.go b/klog.go
index 798f2e3..3f56229 100644
--- a/klog.go
+++ b/klog.go
@@ -778,7 +778,7 @@ func (l *loggingT) errorS(err error, loggr logr.Logger, filter LogFilter, depth
 		msg, keysAndValues = filter.FilterS(msg, keysAndValues)
 	}
 	if loggr != nil {
-		logr.WithCallDepth(loggr, depth+1).Error(err, msg, keysAndValues...)
+		logr.WithCallDepth(loggr, depth+2).Error(err, msg, keysAndValues...)
 		return
 	}
 	l.printS(err, errorLog, depth+1, msg, keysAndValues...)
@@ -790,7 +790,7 @@ func (l *loggingT) infoS(loggr logr.Logger, filter LogFilter, depth int, msg str
 		msg, keysAndValues = filter.FilterS(msg, keysAndValues)
 	}
 	if loggr != nil {
-		logr.WithCallDepth(loggr, depth+1).Info(msg, keysAndValues...)
+		logr.WithCallDepth(loggr, depth+2).Info(msg, keysAndValues...)
 		return
 	}
 	l.printS(nil, infoLog, depth+1, msg, keysAndValues...)
```

Reply:
Yeah, I think this is showing that the current implementation is slightly incorrect as-is

The depth between `InfoDepth` and `InfoSDepth` is different b/c `InfoDepth` goes through one more function call